payjoin/src/core/persist.rs

```diff
@@ -336,14 +336,25 @@ pub enum OptionalTransitionOutcome<NextState, CurrentState> {
     Stasis(CurrentState),
 }
 
+/// Sealed trait to prevent external implementation of SessionEventTrait.
+pub(crate) mod sealed {
+    pub trait Sealed {}
+}
+
+/// Trait for session events that determines if an event represents a session closure.
+pub trait SessionEventTrait: sealed::Sealed {
+    /// Returns true if this event represents a session closure that should close the persister.
+    fn is_closed_event(&self) -> bool;
+}
```

Comment:
The problem seems to be that `HasReplaybleError` events are considered to be "fatal". Could we have a seperate variant (`non_fatal_error` or a better name) on the state transition which when processed 
* Saves an event
* Does not close the session
* Progresses to the error handling typestate (`HasReplyError`) 

I think that means we need a seperate generic (`ReplyState`) along side the others for each state transition object. Unless it can _somehow_ be an associated type since the reply state never changes.